camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEvent.java

```diff
@@ -32,24 +34,63 @@ public WatchMsgEvent(final String _name, final String _session_id, final String
     public void configure() {
         from("timer://foo?synchronous=true&period=" + m_interval)
                 .routeId("manzan_msg:" + m_name)
+                .process(exchange -> {
+                    // Reset the list of ordinal positions at the start of each execution
+                    exchange.setProperty("ordinalPositions", new ArrayList<Integer>());
+                })
                 .setHeader(EVENT_TYPE, constant(ManzanEventType.WATCH_MSG))
                 .setBody(constant("SeLeCt * fRoM " + m_schema + ".mAnZaNmSg wHeRe SESSION_ID = '" + m_sessionId
                         + "' limit " + m_numToProcess))
-                // .to("stream:out")
                 .to("jdbc:jt400?outputType=StreamList")
                 .split(body()).streaming().parallelProcessing()
                 .setHeader("id", simple("${body[ORDINAL_POSITION]}"))
                 .setHeader("session_id", simple("${body[SESSION_ID]}"))
                 .setHeader("data_map", simple("${body}"))
+                .process(exchange -> {
+                    Integer ordinalPosition = exchange.getIn().getHeader("id", Integer.class);
+                    @SuppressWarnings("unchecked")
+                    List<Integer> ordinalPositions = exchange.getProperty("ordinalPositions", List.class);
+
+                    synchronized (ordinalPositions) {
+                        ordinalPositions.add(ordinalPosition); // Accumulate ORDINAL_POSITION
+                    }
+                })
                 .marshal().json(true) // TODO: skip this if we are applying a format
                 .setBody(simple("${body}\n"))
                 .process(exchange -> {
                     if (null != m_formatter) {
                         exchange.getIn().setBody(m_formatter.format(getDataMap(exchange)));
                     }
                 })
-                .recipientList(constant(getRecipientList())).parallelProcessing().stopOnException().end()
-                .setBody(simple("delete fRoM " + m_schema + ".mAnZaNmSg where ORDINAL_POSITION = ${header.id} WITH NC"))
-                .to("jdbc:jt400").to("stream:err");
+                .recipientList(constant(getRecipientList()))
+                .parallelProcessing()
+                .stopOnException()
+                .end()
+                .end()
+                .process(exchange -> {
+                    // Constructing the WHERE clause for ORDINAL_POSITIONs
+                    StringBuilder deleteQuery = new StringBuilder("DELETE FROM " + m_schema
+                            + ".mAnZaNmSg WHERE SESSION_ID = '" + m_sessionId + "' AND ORDINAL_POSITION IN (");
```

Comment:
Not sure why the casing was like this in the first place

```suggestion
                            + ".MANZANMSG WHERE SESSION_ID = '" + m_sessionId + "' AND ORDINAL_POSITION IN (");
```